functions/src/index.ts

```diff
@@ -1,9 +1,115 @@
+            }
+            scheduleExams.push({
+                "name": name,
+                "id": id,
+            });
+        }
+        response.status(200).send({ "data": scheduleExams });
+        console.log("getAllExamsMeta :  data sent");
+    } catch (error) {
+        console.log("getAllExamsMeta: Exception", error);
+
+        response.status(500).send({ "error": "Internal Error" });
+        throw new functions.https.HttpsError("internal", 'Internal Error');
+    }
+});
+
+export const getStudentsOfExam = functions.https.onRequest(async (request, response) => {
+    console.log("getStudentsOfExam : function Start");
+    admin.initializeApp();
+    if (request.body["examID"] == undefined) {
+        console.log("getStudentsOfExam :  missing data error");
+
+        throw new functions.https.HttpsError("invalid-argument", 'Invalid Arguments');
+    }
+    var examsDoc: FirebaseFirestore.QuerySnapshot<FirebaseFirestore.DocumentData>;
+
+    try {
+        examsDoc = await admin.firestore().collectionGroup("exams").where("exam id", '==', request.body["examID"]).get();
+        console.log("getStudentsOfExam :  data fetched");
+
+        var examData = [];
+        for (var i = 0; i < examsDoc.size; i++) {
+            var studentDoc;
+            studentDoc = await examsDoc.docs[i].ref.parent.parent?.get();
+            var studentName = studentDoc?.data["name"];
+            var marksObtained = examsDoc.docs[i].data()['marksObtained'];
+            var maxMarks = examsDoc.docs[i].data()['maxMarks'];
+            examData.push({
+                "studentName": studentName,
+                "marksObtained": marksObtained,
+                "maxMarks": maxMarks,
+            });
+        }
+        console.log("getStudentsOfExam :  data sent");
+
+        response.status(200).send({ "data": examData });
+    } catch (error) {
+        response.status(500).send({ "error": "Internal Error" });
+        console.log("getStudentsOfExam: Exception", error);
+
+        throw new functions.https.HttpsError("internal", 'Internal Error');
+    }
+
+});
+
+
+export const getExamsOfStudent = functions.https.onRequest(async (request, response) => {
+    console.log("getExamsOfStudent :  function Start");
+    admin.initializeApp();
+    if (request.body["studentID"] == undefined) {
+        console.log("getExamsOfStudent :  missing data error");
+
+        throw new functions.https.HttpsError("invalid-argument", 'Invalid Arguments');
+    }
+    var examsDoc: FirebaseFirestore.QuerySnapshot<FirebaseFirestore.DocumentData>;
+    try {
+        examsDoc = await admin.firestore().collection("students").doc(request.body["studentID"]).collection("exams").get();
+        console.log("getExamsOfStudent :  data fetched");
+
+        var examData = [];
+        for (var i = 0; i < examsDoc.size; i++) {
+            var marksObtained = examsDoc.docs[i].data()['marksObtained'];
+            var maxMarks = examsDoc.docs[i].data()['maxMarks'];
```

Comment:
its maxmarks in the firestore doc

{"data":[{"marksObtained":50}]}

the function response is incomplete

Reply:
yes, its a typo mistake
key name in firebase is maxmarks